control-plane/api-gateway/common/resources.go

```diff
@@ -401,6 +402,38 @@ func (s *ResourceMap) ExternalFilterExists(filterRef gwv1beta1.LocalObjectRefere
 	return ok
 }
 
+func (s *ResourceMap) AddGatewayPolicy(gatewayPolicy *v1alpha1.GatewayPolicy) *v1alpha1.GatewayPolicy {
+	sectionName := ""
+	if gatewayPolicy.Spec.TargetRef.SectionName != nil {
```

Comment:
Can any of the parent values to `SectionName` be `nil`? If so, we should `nil` check them to avoid a deref error.

Reply:
I don't believe they can be since Spec and TargetRef are both non nillable.